pcmdi_metrics/variability_mode/variability_modes_driver.py

```diff
@@ -564,9 +564,7 @@
         # try:
         if 1:
             if realization == "*":
-                run = re.split("[._]", (model_path.split("/")[-1]).split("."))[
-                    run_in_modpath
-                ]
+                run = re.split("[._]", model_path.split("/")[-1])[run_in_modpath]
```

Comment:
Shiheng found bug in this line when file is named like “file_name_1.nc”. This fix should work well either cases like file is named like “file_name_1.nc” or “file.name.1.nc”

Reply:
During the mean time, his comment was "I’ve checked the first EOF mode of PDO. The result looks good to me."